hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,233 @@
+Specifically, this means if software issues only WFI, the power manager does not treat it as a power down request.
+The notion of WFI is exported from the processor.
+For Ibex, this is currently in the form of core\_sleeping\_o.
+
+In response to the low power entry request, the fast FSM disables all second level clock gating and asserts appropriate resets if required.  The
+fast FSM then requests the slow FSM to take over.
+Once the request is acknowledged, the fast FSM transitions to low power and waits for the next power up request.
+
+
+## Reset Request Handling
+
+There are 3 reset requests in the system - peripheral requested reset such as watchdog, alert escalation reset and non-debug module reset.
+The flash brownout is handled separately and described in [flash handling section](#Flash-Handling) below.
+
+The watchdog and alert escalation resets are handled directly by the power manager, while the non-debug module reset is handled by the reset controller.
+This separation is because the non-debug reset does not affect the life cycle controller, non-volatile storage controllers and alert states.
+There is thus no need to sequence its operation like the others.
+
+The power controller observes reset requests only in two states - the slow FSM LOW\_POWER state and the fast FSM ACTIVE state - these are considered static states of the system.
+When a reset request is received during slow FSM LOW\_POWER state, the system begins its usual power up sequence even if a wakeup has not been received.
+
+When a reset request is received during fast FSM ACTIVE state, either because the system was already active or exited low power due to the reset request,  the fast FSM asserts resets immediately and transitions back through its initialization process.
+Note in this case, the slow FSM is “not activated” and remains in its IDLE state.
+
+
+### Reset Requests Received During Other States
+
+All other states in the slow / fast FSM are considered transition states.
+Resets are not observed in other states because the system will always be transitioning towards one of the steady states (the system is in the process of powering down or powering up).
+Once a steady state is reached, the reset request is then observed and action taken.
+
+### Reset Recording
+
+Of the resets directly handled by the power manager (peripheral reset request, escalation and low power entry), the power manager sequences only 1 at a time (see state diagrams).
+This means if all reset requests happen at approximately the same time, the power manager will handle in a first come first serve manner.
+When the handling of the first is completed, it observes there is another pending request and handles accordingly.
+This is done because low power resets and peripheral requested resets lead to different behaviors.
+When the power manager commits to handling a specific request, it sends a reset hint to the reset manager to inform it of what reasons to record should the processor reset.
```

Comment:
This feels a little hard to parse. Maybe something like this would be better? "... it sends a reset hint to the reset manager, informing it what reasons to record if it resets the processor."

Reply:
i reworded this